libr/util/charset.c

```diff
@@ -113,24 +113,33 @@ R_API size_t r_charset_encode_str(RCharset *rc, ut8 *out, size_t out_len, const
 
 // assumes out is as big as in_len
 R_API size_t r_charset_decode_str(RCharset *rc, ut8 *out, size_t out_len, const ut8 *in, size_t in_len) {
-	char k[32];
 	char *o = (char*)out;
-	int i;
-	for (i = 0; i < in_len; i++) {
-		ut8 ch_in = in[i];
-
-		//zero terminate the string
-		snprintf (k, sizeof (k), "%c", ch_in);//snprintf (k, sizeof (k), "0x%02x", ch_in);
-		char *v = sdb_get (rc->db_char_to_hex, k, 0);
-		memmove(v, v+2, strlen (v));
-
-		//convert to ascii
-		char str_hx[32];
-		snprintf (str_hx, sizeof (str_hx), "%c", (char) strtol( v, 0, 16));
-		const char *ret = r_str_get_fail (str_hx, "?");
-		strcpy (o, ret);
-
-		o += strlen (o);
+	size_t last_char_size;
+	size_t maxkeylen = 8;
+
+	size_t cur, j;
+	for (cur = 0; cur < in_len; ) {
+		char *str = r_str_ndup((char *)in, maxkeylen);
```

Comment:
shouldnt it be in + cur, R_MIN (maxkeylen, in_len-cur)) ?

Reply:
It works like that...

Reply:
Yes. It works because in is luckily null terminated. But it is not correct